tests/federation/test_printer.py

```diff
@@ -66,3 +66,74 @@ def top_products(self, first: int) -> List[Product]:
     assert schema.as_str() == textwrap.dedent(expected).strip()
 
     del Review
+
+
+def test_fields_requires_are_printed_correctly():
+    global Review
+
```

Comment:
This is the only test I can think of, I am not sure how to test it better without having a federation server and so on?